### PR TITLE
Improve UI & Cowebsites

### PR DESCRIPTION
**Pull request by nolway**
CoWebsites V1 + Improve UI management.

### CODE REVIEW

File: front/src/Phaser/Game/GameScene.ts [hunk 1]
```diff
@@ -2063,28 +2075,21 @@ ${escapedMessage}
...
-        this.connection?.setSilent(!!silent);
-        jitsiFactory.stop();
-        mediaManager.showGameOverlay();
-
-        mediaManager.removeTriggerCloseJitsiFrameButton("close-jitsi");
+        const coWebsite = coWebsiteManager.searchJitsi();
+        if (coWebsite) {
+            coWebsiteManager.closeCoWebsite(coWebsite).catch(() => {
+                console.error("Error during Jitsi co-website closing");
+            });
```
Comment (moufmouf) [inline front/src/Phaser/Game/GameScene.ts:2091]:
```suggestion
            coWebsiteManager.closeCoWebsite(coWebsite).catch((e) => {
                console.error("Error during Jitsi co-website closing", e);
            });
```

Let's try to keep the original error message

File: front/src/Components/MainLayout.svelte [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+    @import "../../style/breakpoints.scss";
+
+    #main-layout {
+        display: grid;
+        grid-template-columns: 7% 93%;
+        grid-template-rows: 80% 20%;
+
+        &-left-aside {
+            min-width: 80px;
+        }
```
Comment (moufmouf) [inline front/src/Components/MainLayout.svelte:125]:
```suggestion
    #main-layout {
        display: grid;
        grid-template-columns: 80px calc(100% - 80px);
        grid-template-rows: 80% 20%;

        &-left-aside {
            width: 80px;
        }
```

Pour éviter que les 7% ne soient < à 80px et que les iframe overlap avec les boutons

File: front/src/Components/EmbedScreens/Layouts/MozaicLayout.svelte [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    >
+        {#each [...$streamableCollectionStore.values()] as peer (peer.uniqueId)}
+            <MediaBox
+                streamable={peer}
+                mozaicFullWidth={$streamableCollectionStore.size === 1 || $streamableCollectionStore.size === 2}
+                mozaicQuarter={$streamableCollectionStore.size === 3 || $streamableCollectionStore.size === 4}
+            />
+        {/each}
+    </div>
+</div>
```
[the remark this answers is not in the document]
Comment (nolway) [reply front/src/Components/EmbedScreens/Layouts/MozaicLayout.svelte:31]: The defaut css is for more than 4 streamables.
The mozaic layout cannot contain a cowebsite only the the main is displayed.

File: front/src/Messages/generated/.gitignore [hunk 4]
```diff
@@ -0,0 +1,2 @@
+*
+!.gitignore
```
Comment (moufmouf) [inline front/src/Messages/generated/.gitignore:2]: You can remove this file

**Comment by moufmouf** [comment]
One last very small improvement:

When the icons are loading from the icon server, the icon is displayed before it is actually loaded. I think the "alt" text is displayed instead at this point.

https://user-images.githubusercontent.com/1290952/151363010-45ca5dc4-e177-4ea1-971b-634eb75cbf02.mp4

This is a normal behaviour on the web, but not so much for a game. I think it would be better to have either a placeholder (like in https://stackoverflow.com/questions/32909488/how-do-i-make-a-placeholder-image-in-html-if-the-original-image-hasnt-been-foun) or to use display the button only when the image is actually loaded.

**Comment by Lurkars** [comment]
I must complain, that I don't understand this change in removing the websiteTriggerMessage completely. First of all, I just liked the feature as it was, also it's in line with jitsi triggers etc. But what is biggest case for me is, that I removed the Icon feature for Co-Website due to privacy reasons. But with this, I need to add a generic icon which then gives no context at all about the website to open. Also there may be sites offering generic/not easy relatable icons, so even with icon I have no hint about what will be opened on click. So first question, is there any chance in putting back the triggerMessages? And if not, I really would appreciate an explanation about the decision about removing this. @nolway @moufmouf

**Comment by moufmouf** [comment]
@Lurkars can you open a separate issue with your message above? So we can keep track of it more easily than in a closed PR?

**Pull request merged**